frontends/parsers/p4/p4parser.ypp

```diff
@@ -556,12 +556,13 @@ functionPrototype
     ;
 
 methodPrototype
-    : functionPrototype ";" { $$ = $1; }
-    | ABSTRACT functionPrototype ";"    { $$ = $2; $$->setAbstract(); }  // experimental
-    | TYPE "(" parameterList ")" ";"  // constructor
-                                        { auto par = new IR::ParameterList(@3, *$3);
-                                          auto mt = new IR::Type_Method(@1, par);
-                                          $$ = new IR::Method(@1, IR::ID(@1, $1), mt); }
+    : optAnnotations functionPrototype ";" { $$ = $2; $2->annotations = $1; }
+    | optAnnotations ABSTRACT functionPrototype ";"    { $$ = $3; $$->setAbstract();
+                                                         $3->annotations = $1; }  // experimental
+    | optAnnotations TYPE "(" parameterList ")" ";"  // constructor
```

Comment:
I don't know if annotations on the constructor are useful; you already have annotations on the type declaration.

Reply:
In fact, you are not using the annotations here.

Reply:
Annotation on constructor is used in the 4th argument in `new IR::Method()`

Reply:
Without the `optAnnotations` on the constructor, there is a shift/reduce conflict.